From ec960ed1dc64ae3513167687141ea48c7cec5ebd Mon Sep 17 00:00:00 2001
From: "Vladimir D. Seleznev" <vseleznv@altlinux.org>
Date: Mon, 15 Apr 2019 02:33:09 +0300
Subject: [PATCH 07/13] ALT TCL_LIBS for shared linkage

Add THREADS_LIBS to TCL_LIB_FLAG, tclConfig.sh provides flags for shared
linkage.
---
 tcl/unix/configure.in    | 4 ++--
 tcl/unix/tclConfig.sh.in | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/tcl/unix/configure.in b/tcl/unix/configure.in
index 08b92717eb..54d1fa6a6c 100755
--- a/tcl/unix/configure.in
+++ b/tcl/unix/configure.in
@@ -841,9 +841,9 @@ else
     # default install directory for bundled packages
     PACKAGE_DIR="$libdir"
     if test "${TCL_LIB_VERSIONS_OK}" = "ok"; then
-        TCL_LIB_FLAG="-ltcl${TCL_VERSION}"
+        TCL_LIB_FLAG="-ltcl${TCL_VERSION} $THREADS_LIBS"
     else
-        TCL_LIB_FLAG="-ltcl`echo ${TCL_VERSION} | tr -d .`"
+        TCL_LIB_FLAG="-ltcl`echo ${TCL_VERSION} | tr -d .` $THREADS_LIBS"
     fi
     TCL_BUILD_LIB_SPEC="-L`pwd | sed -e 's/ /\\\\ /g'` ${TCL_LIB_FLAG}"
     TCL_LIB_SPEC="-L${libdir} ${TCL_LIB_FLAG}"
diff --git a/tcl/unix/tclConfig.sh.in b/tcl/unix/tclConfig.sh.in
index b8aa935323..a116b0e927 100644
--- a/tcl/unix/tclConfig.sh.in
+++ b/tcl/unix/tclConfig.sh.in
@@ -40,7 +40,7 @@ TCL_SHARED_BUILD=@TCL_SHARED_BUILD@
 TCL_LIB_FILE='@TCL_LIB_FILE@'
 
 # Additional libraries to use when linking Tcl.
-TCL_LIBS='@TCL_LIBS@'
+TCL_LIBS='@TCL_LIB_FLAG@ @TCL_STUB_LIB_FLAG@'
 
 # Top-level directory in which Tcl's platform-independent files are
 # installed.
-- 
2.24.0

